packages/dd-trace/src/plugins/util/test.js

```diff
@@ -69,7 +69,10 @@ function getTestParametersString (parametersByTestName, testName) {
 }
 
 function finishAllTraceSpans (span) {
-  span.context()._trace.started.forEach((span) => {
+  const spansToFinish = span.context()._trace.started.filter(traceSpan =>
+    !traceSpan.context()._isFinished && span !== traceSpan
+  )
+  spansToFinish.forEach(span => {
```

Comment:
I was having weird sort of errors finishing spans more than once. Also, letting the test span finish normally seems to work better too. I couldn't find the source of the problem, but this just works better

Reply:
This is really odd as calling `finish` on a finished span is a no-op, so it might be something to do with the test span as you mentioned. Can you look into it a bit more since the issue seems to be specific to the test span?